docs/source/grouping.md

```diff
@@ -0,0 +1,62 @@
+<!--- Copyright (c) 2024, NVIDIA CORPORATION.
+SPDX-License-Identifier: BSD-3-Clause -->
+
+# Grouping
+
+Grouping allows for rule-based batching of samples into one batch on the fly. 
+
+Note how this is different from [packing](packing) which is done *in addition* to batching (it's done before batching) and
+joins multiple samples into one.
+On the other hand grouping *replaces* standard batching.
```

Comment:
I wouldn't say it "replaces" standard batching. Because from a user perspective, the "standard batching" still looks the same? (i.e. `batch` function still works as expected). The difference is, that items are grouped now 🤷 
I'd rather write here, when to use packing VS when to use grouping?